named_entity_recognition_bert/named_entity_recognition_huggingface_bert_pipeline.ipynb

```diff
@@ -0,0 +1,1124 @@
+    "\n",
+    "    train_dataset = DataSequence(train_texts, train_entities, label2id)\n",
+    "    val_dataset = DataSequence(val_texts, val_entities, label2id)\n",
+    "\n",
+    "    train_dataloader = torch.utils.data.DataLoader(train_dataset,\n",
+    "                                                   num_workers=4,\n",
+    "                                                   batch_size=BATCH_SIZE,\n",
+    "                                                   shuffle=True)\n",
+    "    \n",
+    "    val_dataloader = torch.utils.data.DataLoader(val_dataset,\n",
+    "                                                 num_workers=4,\n",
+    "                                                 batch_size=BATCH_SIZE)\n",
+    "    \n",
+    "    train_size = len(train_texts)\n",
+    "    val_size = len(val_texts)\n",
+    "\n",
+    "    use_cuda = torch.cuda.is_available()\n",
+    "    device = torch.device(\"cuda\" if use_cuda else \"cpu\")\n",
+    "    \n",
+    "    optimizer = NAdam(model.parameters(), lr=LEARNING_RATE)\n",
+    "\n",
+    "    if use_cuda:\n",
+    "        model = model.cuda()\n",
+    "\n",
+    "    best_acc = 0\n",
+    "    best_loss = 1000\n",
+    "\n",
+    "    for epoch_num in range(EPOCHS):\n",
+    "\n",
+    "        total_acc_train = 0\n",
+    "        total_loss_train = 0\n",
+    "\n",
+    "        model.train()\n",
+    "\n",
+    "        for train_data, train_label in tqdm(train_dataloader):\n",
+    "            train_label = train_label.to(device)\n",
+    "            mask = train_data['attention_mask'].squeeze(1).to(device)\n",
+    "            input_id = train_data['input_ids'].squeeze(1).to(device)\n",
+    "\n",
+    "            optimizer.zero_grad()\n",
+    "            loss, logits = model(input_id, mask, train_label)\n",
+    "\n",
+    "            for i in range(logits.shape[0]):\n",
+    "\n",
+    "                logits_clean = logits[i][train_label[i] != PAD_TOKEN_ID]\n",
+    "                label_clean = train_label[i][train_label[i] != PAD_TOKEN_ID]\n",
+    "\n",
+    "                predictions = logits_clean.argmax(dim=1)\n",
+    "                acc = (predictions == label_clean).float().mean()\n",
+    "                total_acc_train += acc\n",
+    "                total_loss_train += loss.item()\n",
+    "\n",
+    "            loss.backward()\n",
+    "            optimizer.step()\n",
+    "\n",
+    "        model.eval()\n",
+    "\n",
+    "        total_acc_val = 0\n",
+    "        total_loss_val = 0\n",
+    "\n",
+    "        for val_data, val_label in val_dataloader:\n",
+    "\n",
+    "            val_label = val_label.to(device)\n",
+    "            mask = val_data['attention_mask'].squeeze(1).to(device)\n",
+    "            input_id = val_data['input_ids'].squeeze(1).to(device)\n",
+    "\n",
+    "            loss, logits = model(input_id, mask, val_label)\n",
+    "\n",
+    "            for i in range(logits.shape[0]):\n",
```

Comment:
Same part for train and validation move to some method of class
          
           val_label = val_label.to(device)
            mask = val_data['attention_mask'].squeeze(1).to(device)
            input_id = val_data['input_ids'].squeeze(1).to(device)

            loss, logits = model(input_id, mask, val_label)

            for i in range(logits.shape[0]):
                logits_clean = logits[i][val_label[i] != PAD_TOKEN_ID]
                label_clean = val_label[i][val_label[i] != PAD_TOKEN_ID]

                predictions = logits_clean.argmax(dim=1)
                acc = (predictions == label_clean).float().mean()